docs/browser-usage.md

```diff
@@ -22,17 +24,38 @@ module.exports = {
 
 In case of cypress, you can adapt the same webpack config trick just by applying the [cypress-webpack-preprocessor](https://github.com/cypress-io/cypress-webpack-preprocessor) plugin.
 
+## Angular and similar
+
 However, in some TypeScript projects like the ones using Angular, which AoT compiler requires that a full `*.ts` file is provided instead of just a `*.js` and `*.d.ts` files, to use this shim we have to simply set up our TypeScript configuration in `tsconfig.json` to use this file instead of a normal TypeGraphQL module:
 
 ```json
 {
   "compilerOptions": {
     "baseUrl": ".",
     "paths": {
-      "type-graphql": ["./node_modules/type-graphql/dist/browser-shim.ts"]
+      "type-graphql": ["node_modules/type-graphql/dist/browser-shim.ts"]
+    }
+  }
+}
+```
+
+## Next.js and similar
+
+When using the shim with Next.js as a dedicated frontend server be aware that Next has pre-renders on the server. This means that in development mode the `webpack: {}` config in `next.config.js` is skipped and full `type-graphql` is bundled. But you still need to handle some webpack rewiring for the client bundling which still happens with webpack both in development and in production mode.
+
+The easiest way is to accomplish this is also done in  `tsoconfig.json`—add the same keys like in the example before to `compilerOptions`:
+
+```json
+{
+  "compilerOptions": {
+    "baseUrl": ".",
+    "paths": {
+      "type-graphql": ["node_modules/type-graphql/dist/browser-shim.ts"]
     }
   }
 }
 ```
 
+Then, `npm install -D tsconfig-paths` and enable it with `NODE_OPTIONS="-r tsconfig-paths/register"` in your environment variables setup.
```

Comment:
I think it's too detailed - the CRA section doesn't show how to install and configure the `craco` package

Reply:
Tbh, I would even elaborate more on this.

Next is quite sophisticated on how it does everything and it is by far the most used framework (CRA and Angular are a fraction of Next). You should def keep this snippet, otherwise your user won't get Next to work:

- Next's webpack config automatically pulls any `tsconfig` `paths` redirection and apply it to its compile step without the need to any webpack plugin like usually (which is great)
- However, you need to apply this as well to the runtime which is done by `node` which is hidden and abstracted away under the `next` CLI
- Since you can't access the underlying `node` you need to pass `tsconfig-paths/register` via the env `NODE_OPTIONS`

Once you now above life is easy but to get there took me +2 days and again since Next is the major React platform this README should focus on Next. The rest is nice-to-have but tbh, nobody cares.

Re craco, you should def add that the CRA scripts do not work anymore and craco's replacement script need to be used. 

Let me know how you want to the shim doc and i can rewrite all.

Reply:
Or what we could also do, just forget all the framework-specific crap and educate the user and tell him that he need redirects for compile-time and build-time and depending on his framwork his does this with webpack-plugins, ts paths config and/or the tsconfig-paths module

Reply:
i am vetting out next.js and type-graphql combo. After scouring for docs + tutorials + videos, I have a relevant perspective on this. In my personal opinion, the more detailed the better? It would definitely increase the dev ux which arguably increases adoption. I tried nexus then type-graphql, nexus has weird typescript errors that are explained in different places which made the whole setup rather painful, I have since abandoned the nexus route because typegraphql docs got me farther.